core-ui/src/components/Extensibility/components-form/ResourceRenderer.js

```diff
@@ -31,30 +31,49 @@ export function ResourceRenderer({
     scope === 'namespace' ? namespace : null,
   );
 
-  const { data } = useGetList()(url);
-
-  const options = (data || []).map(res => ({
-    key: res.metadata.name,
-    text: res.metadata.name,
-  }));
+  let expression;
+  if (schema.get('filter')) {
+    expression = jsonataWrapper(schema.get('filter'));
+    expression.assign('root', props?.resource);
+  }
 
   return (
     <ResourceForm.FormField
-      value={value}
-      setValue={value => {
-        onChange({
-          storeKeys,
-          scopes: ['value'],
-          type: 'set',
-          schema,
-          required,
-          data: { value },
-        });
-      }}
       label={tFromStoreKeys(storeKeys, schema)}
       data-testid={storeKeys.join('.')}
-      input={Inputs.ComboboxInput}
-      options={options}
+      input={() => (
+        <K8sResourceSelectWithUseGetList
+          url={url}
+          filter={item => {
+            if (expression) {
+              expression.assign('item', item);
+              return expression.evaluate();
+            } else return true;
+          }}
+          onChange={value => {
+            onChange({
+              storeKeys,
+              scopes: ['value'],
+              type: 'set',
+              schema,
+              required,
+              data: { value },
+            });
+          }}
+          onSelect={value => {
+            onChange({
+              storeKeys,
+              scopes: ['value'],
+              type: 'set',
+              schema,
+              required,
+              data: { value },
+            });
+          }}
```

Comment:
These callbacks are the same, please move them to a single function.

Reply:
The same goes for `core-ui/src/components/Extensibility/components-form/ResourceRenderer.js`